lib/client.js

```diff
@@ -1000,28 +1174,43 @@ MqttClient.prototype._handleAck = function (packet) {
     case 'pubcomp':
       // same thing as puback for QoS 2
     case 'puback':
+      var pubackRC = packet.responseCode
       // Callback - we're done
       delete this.outgoing[mid]
       this.outgoingStore.del(packet, cb)
+      if (pubackRC && pubackRC > 0 && pubackRC !== 16) {
+        err = new Error('Publish error: ' + errors[pubackRC])
+        err.code = pubackRC
+        this.emit('error', err)
```

Comment:
we should call the callback with the error, not emitting one.

Reply:
rebase to call callback